File: PAT/BLevel/1013/1013.cpp
// 1013 数素数
// auguralpha
// 2019-02-25

#include <cstdio>
#include <vector>
using namespace std;
vector<int> vec;

bool is_prime(int n);

int main() {
    int m, n;
    int cnt = 0;
    scanf("%d%d", &m, &n);
    for(int i = 2; ; ++ i) {
        if(is_prime(i)) {
            ++ cnt;
            if(m <= cnt && n >= cnt) {
                vec.push_back(i);
            }
        }
        if(n < cnt) {
            break;
        }
    }
    for(int i = 0; i < vec.size(); ++ i) {
        if(i % 10 == 9 || i == vec.size() - 1) {
            printf("%d\n", vec[i]);
        }
        else {
            printf("%d ", vec[i]);
        }
    }
    return 0;
}

bool is_prime(int n) {
    for(int i = 2; i * i <= n; ++ i) {
        if(n % i == 0) {
            return false;
        }
    }
    return true;
}

File: PAT/BLevel/1013/1013.go
// 1013 数素数
// auguralpha
// 2020-03-25

package main

import "fmt"

// 判断素数
func isPrime(n int) bool {
	for i := 2; i * i <= n; i  ++ {
		if n % i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var m, n int
	var cnt int
	var ans []int
	_, _ = fmt.Scanf("%d%d", &m, &n)
	for i := 2; ; i ++ {
		if isPrime(i) {
			cnt ++
			if m <= cnt && n >= cnt {
				ans = append(ans, i)
			}
		}

		if n < cnt {
			break
		}
	}

	for i := 0; i < len(ans); i ++ {
		if i % 10 == 9 || i == len(ans) - 1 {
			fmt.Printf("%d\n", ans[i])
		} else {
			fmt.Printf("%d ", ans[i])
		}
	}
}
